tests/Serializer/AbstractItemNormalizerTest.php

```diff
@@ -835,8 +835,6 @@ public function testChildInheritedProperty()
 
         $serializerProphecy = $this->prophesize(SerializerInterface::class);
         $serializerProphecy->willImplement(NormalizerInterface::class);
-        $serializerProphecy->normalize('foo', null, Argument::type('array'))->willReturn('foo')->shouldBeCalled();
-        $serializerProphecy->normalize(null, null, Argument::type('array'))->willReturn(null)->shouldBeCalled();
```

Comment:
but now this may break some stuff no? :p

Reply:
See OP:

> Potential BC breaks:
> 
> - If anyone has been using custom normalizers for null and scalar values, those custom normalizers will no longer be called. I think that is really an edge case, while the performance win of this change would be felt by the majority of our users.